src/scancode/cli.py

```diff
@@ -223,20 +232,39 @@ def reindex_licenses(ctx, param, value):
 number of files processed. Use --verbose to display file-by-file progress.
 '''
 
+formats = ('json', 'json-pp', 'html', 'html-app', 'spdx-tv', 'spdx-rdf')
+plugin_formats = {}
 
 class ScanCommand(BaseCommand):
     short_usage_help = '''
 Try 'scancode --help' for help on options and arguments.'''
 
+    options = [ op for op in scan_proper_pm.hook.add_cmdline_option() if isinstance(op, click.Option) ]
+
+    def __init__(self, name, context_settings=None, callback=None,
+                 params=None, help=None, epilog=None, short_help=None,
+                 options_metavar='[OPTIONS]', add_help_option=True):
+        BaseCommand.__init__(self, name, context_settings, callback,
+                 params, help, epilog, short_help,
+                 options_metavar, add_help_option)
+        plugins_ret_list = scan_proper_pm.hook.add_format()
+        for plugin_ret in plugins_ret_list:
+            if (len(plugin_ret) == 2 and plugin_ret[0] not in plugin_formats):
+                plugin_formats[plugin_ret[0]] = plugin_ret[1]
+        if plugin_formats:
+            for param in self.params:
+                if param.human_readable_name == 'format':
+                    param.help = param.help.replace(
```

Comment:
This looks weird to have a whart for custom formats: I reckon you are trying to keep the same CLI UX, but it might be cleaner to update the CLI have a separate option for `--format-custom <path to format template` and remove this kludge